include/pagmo/algorithms/gaco.hpp

```diff
@@ -0,0 +1,978 @@
+                            temporary_archive[i][1 + variables[0].size() + i_obj] = fitness[jj][i_obj];
+                        }
+                        count_2 = true;
+                    }
+                }
+            }
+
+            // I increase the evalstop parameter only if the best solution of the archive is updated:
+            if (sol_archive[0][0] == temporary_archive[0][0]) {
+                ++n_evalstop;
+
+            } else {
+                n_evalstop = 1;
+            }
+
+            // I hereby update the solution archive:
+            for (decltype(m_ker) i = 0u; i < m_ker; ++i) {
+                sol_archive[i] = temporary_archive[i];
+            }
+        } else {
+            ++n_impstop;
+            ++n_evalstop;
+        }
+        if (n_evalstop == 1 || n_evalstop > 2) {
+            ++gen_mark;
+        }
+        if (gen_mark > m_n_gen_mark) {
+            gen_mark = 1;
+        }
+    }
+
+    /**
+     * Function which computes the pheromone values (useful for generating offspring)
+     *
+     * @param[in] gen_mark Generation mark: this parameter is used to reduce the standard deviation
+     * @param[in] gen Generations: current generation number is passed
+     * @param[in] prob_cumulative Cumulative probability vector: this vector will be crucial for the new individuals'
+     * creation
+     * @param[in] omega_vec Omega: the weights are passed to be modified (i.e., they are one of the pheromone values)
+     * @param[in] sigma_vec Sigma: the standard deviations are passed to be modified (i.e., they are one of the
+     * pheromone values)
+     * @param[in] popul Population: the current population is passed
+     * @param[in] sol_archive Solution archive: the solution archive is useful for retrieving the current individuals
+     * (which will be the means of the new pdf)
+     */
+    void pheromone_computation(unsigned gen_mark, const unsigned gen, vector_double &prob_cumulative,
+                               vector_double &omega_vec, vector_double &sigma_vec, const population &popul,
+                               std::vector<vector_double> &sol_archive) const
+    {
+
+        const auto &prob = popul.get_problem();
+        const auto bounds = prob.get_bounds();
+        const auto &lb = bounds.first;
+        const auto &ub = bounds.second;
+        auto n_con = prob.get_nx();
+
+        // Here I define the weights. Their definition depends on the user's selection of the m_omega_strategy parameter
+        // (either 1 or 2)
+        if (gen == 1 && m_omega_strategy == 1) {
+            // We declare a vector with 'ker' doubles
+            vector_double ker_vector(m_ker);
+            // We fill it with 1,2,3,...,K
+            std::iota(std::begin(ker_vector), std::end(ker_vector), 1);
+            // We compute the sum of the elements
+            double sum = std::accumulate(ker_vector.begin(), ker_vector.end(), 0);
```

Comment:
This is a tricky one (and the reason for the good practice of using ```dots``` for double constants ```u``` for unsigned etc...). When you us ```std::accumultate``` and you init with the value ```0```, the compiler deduces its type as integer. He will then start accumulating integers and will return an integer (https://en.cppreference.com/w/cpp/algorithm/accumulate), while ker_vector is full of doubles, so he converts your doubles to integers (https://ci.appveyor.com/project/ci4esa/pagmo2/builds/22527804/job/ls60fetvov7v5m4u#L934). 

This is a plain bug as I guess that is not what you want to do!!

Reply:
Fortunately, I was summing integers even though I defined a vector_double as ker_vector. Thus, the results were not affected by this. I nevertheless fixed this bug by substituting `0` with `0u` as you suggested, and `vector_double ker_vector(m_ker)` with `std::vector <unsigned> ker_vector(m_ker)`.

Reply:
you may still have issues as sum is ```double``` and some compilers do not like to convert implicitly ints do doubles ... but lets see

Reply:
Maybe defining `sum` as `unsigned` will fix the issue